server/policies/document.ts

```diff
@@ -177,6 +177,45 @@ allow(User, ["pin", "unpin"], Document, (user, document) => {
   return user.teamId === document.teamId;
 });
 
+allow(User, ["subscribe", "unsubscribe"], Document, (user, document) => {
+  // Sanity check.
+  if (!document) {
+    return false;
+  }
+
+  // Subscribing to archived document doesn't make sense.
+  // Unarchiving it should resume subscriptions on change.
+  if (document.archivedAt) {
+    return false;
+  }
+
+  if (document.deletedAt) {
+    return false;
+  }
+
+  if (document.template) {
+    return false;
+  }
+
+  if (!document.publishedAt) {
+    return false;
+  }
+
+  // Sanity check.
+  invariant(
+    document.collection,
+    "Collection is missing, did you forget to include in the query scope?"
+  );
+
+  if (cannot(user, "update", document.collection)) {
```

Comment:
```suggestion
  if (cannot(user, "read", document.collection)) {
```

Don't need edit permissions to subscribe, only read permissions